code/datums/components/wall_mounted.dm

```diff
@@ -0,0 +1,72 @@
+// This element should be applied to wall-mounted machines/structures, so that if the wall it's "hanging" from is broken or deconstructed, the wall-hung structure will deconstruct.
+/datum/component/wall_mounted
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	/// The object that is currently linked to the wall.
+	var/obj/linked_object
+	/// Callback to the parent's proc to call on the linked object when the wall disappear's or changes.
+	var/datum/callback/on_drop
```

Comment:
for that reason I really think this component should be applied to the object, rather then the turf.
it allows us to track object movement with more detail in future, and just makes more sense in terms of what's going on.